### PR TITLE
Improve retry logic and add python 3 classifier

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -15,6 +15,7 @@ classifier =
     License :: OSI Approved :: BSD License
     Operating System :: OS Independent
     Programming Language :: Python
+    Programming Language :: Python :: 3
```
Comment (vidartf) [inline setup.cfg:18]: Perhaps we should use `python_requires` instead to ensure we do not break the install of someone on Python 2? (https://stackoverflow.com/questions/42238484/prevent-package-from-being-installed-on-old-python-versions)

Comment (blink1073) [reply setup.cfg:18]: Happy to accept a PR ;)

Comment (vidartf) [reply setup.cfg:18]: Sure, just double checking that I wasn't missing some argument :)

Comment (blink1073) [reply setup.cfg:18]: IMHO Python 2 is a dead language and you can pin if you need to.

Comment (vidartf) [reply setup.cfg:18]: That is of course an acceptable answer as well :)

**Pull request merged**